### DISCUSSION TITLE
How to display displacement?

### DISCUSSION THREAD

**Discussion by zhang-rite** [idaholab/moose#17122] [Q&A Modules: General] [answered]
Hi 

Here is a native question. My problem has primary variables "disp_r disp_z". In default output exodus file, there shows a variable of "disp_". Now I want to see the differential displacement of each step to a special time (_initial). 
The questions are 
- How to make "disp_r" and "disp_z" can be visualized separately by exodus file?
- **how can I define "disp_r_initial" in auxkernel or in kernel or in some other ways** then I can calculate the difference? 

I know the way for strain and stress. I just don't know how to do this for primary variables. Thank you.

**Comment by WilkAndy** [comment, marked as answer]
Seems like i'm your personal assistant!  :-)

1. This is a paraview question.  Look up the top of the paraview window.  You should see a box containing a list of variables you can display.  One of them will be "disp_" as you have noticed already.  Click it.  Then look to the right of this box, and you should see a box with "Magnitude; x; y; z" in it.  Choose your desired component (your disp_r will be called "x" by paraview)

2. Use `ParsedAux` and execute on initial.  Eg

```
 [disp_r_initial]
    type = ParsedAux
    args = disp_r
    function = disp_r
    variable = disp_r_initial
    execute_on = initial
  []
 ```

You can also define

```
 [disp_r_change]
    type = ParsedAux
    args = 'disp_r disp_r_initial'
    function = 'disp_r - disp_r_initial'
    variable = disp_r_change
 []
 ```

**Reply by zhang-rite** [reply]
Wow, Andy, thanks again and again :smile: !
